blocks/api/categories.js

```diff
@@ -27,3 +30,104 @@ const categories = [
 export function getCategories() {
 	return categories;
 }
+
+/**
+ * Register a new block category
+ *
+ * @param {Array} cat e.g {slug: 'custom', title: __('Custom Blocks')}
+ *
+ * @return {Array} Block categories
+ *
+ */
+export function registerCategory( cat ) {
+	if ( ! cat ) {
+		console.error(
+			'The Block category must be defined'
+		);
+		return;
+	}
+	if ( ! cat.slug ) {
+		console.error(
+			'The Block category slug must be defined'
+		);
+		return;
+	}
+	if ( ! /^[a-z0-9-]+$/.test( cat.slug ) ) {
+		console.error(
+			'Block category slug must not contain characters which are invalid for urls'
+		);
+		return;
+	}
+	if ( categories.find( x => x.slug === cat.slug ) ) {
+		console.error(
+			'Block category "' + cat.slug + '" is already registered'
+		);
+		return;
+	}
+	if ( ! cat.title ) {
+		console.error(
+			'The Block category title must be defined'
+		);
+		return;
+	}
+
+	categories.push( cat );
+	return categories;
+}
+
+/**
+ *
+ * Sort categories by key
+ *
+ * @param {String} key The key to sort by
+ *
+ * @returns {Array} Block categories
+ */
+export function sortCategoriesBy( key ) {
+	if ( ! key ) {
+		console.error(
+			'The key must be defined'
+		);
+		return;
+	}
+	if ( typeof key !== 'string' ) {
+		console.error(
+			'The key must be a string'
+		);
+		return;
+	}
+	categories = sortBy( categories, key );
+	return categories;
+}
+
+/**
+ * Set the property 'order' for a category
+ *
+ * @param {String}    slug    The slug for the category
+ * @param {Number}    order   The order for the category
+ *
+ * @returns {Array} Block categories
+ */
+export function setCategoryOrder( slug, order ) {
+	const pos = findIndex( categories, ( category ) => category.slug === slug );
```

Comment:
Lodash's [`_.find`](https://lodash.com/docs/4.17.4#find) might be closer to what you want, since you don't really care about the index aside from referencing the matched object itself.